include/vrv/layerelement.h

```diff
@@ -467,6 +467,22 @@ class LayerElement : public Object,
 private:
     int GetDrawingArticulationTopOrBottom(data_STAFFREL place, ArticType type);
 
+    /**
+     * Template for getting above/below overflow for BeamType elements (BEAM or FTREM)
+     */
+    template <class BeamType> void GetBeamOverflow(StaffAlignment *&above, StaffAlignment *&below);
+
+    /**
+     * Template for getting above/below overflow for children of BeamType elements (BEAM or FTREM)
+     */
+    template <class BeamType>
+    void GetBeamChildOverflow(StaffAlignment *&above, StaffAlignment *&below, BeamType *parent);
+
```

Comment:
I am not to fond of having beam related methods with templates in LayerElement. The BeamInterface is there for not having to do this. If you need the LayerElement in some methods, you can pass it as parameter to the BeamInterface method. I would agree that this is a bit weird, but we have that already elsewhere (e.g.: ObjectListInterface::GetList(Object *)`. The main advantage would be not to overload LayerElement and to keep the beam related code in one place.

Reply:
Never mind, we can look at that later